lib/order_list.rb

```diff
@@ -0,0 +1,45 @@
+class OrderList 
+  attr_reader :orders
+
+  def initialize 
+    @orders = []
+  end
+
+  def receive_order(order)
+    orders.push order
+    self
+  end
+
+  def print value=nil
+    { order: { items: itemized_list, total: total.round(2) }}
+  end
+
+  def total
+    orders.inject(0) {|sum, order| sum + order.cost }
+  end
+  
+  def itemized_list 
+    condense orders.map(&:print)
+  end
+
+  def condense order_hashes
+    order_hashes = condense_by_quantity order_hashes
```

Comment:
Took me a while to work out what's going on here, again I think there's a simpler solution that would avoid all this complexity - if you have the orders as a hash and name as a key, you could add additional orders to that order rather than have them all floating as seperate items in an array that you then have to tidy up here.